src/core/vector/qgsvectorlayer.h

```diff
@@ -824,10 +824,14 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      * \param ids feature IDs to select
      * \param behavior selection type, allows adding to current selection, removing
      * from selection, etc.
+     * \param validateIds if TRUE, filters out invalid feature IDs before
+     * selection. Set to FALSE (default) for best performance. Enable validation when calling from
+     * Python or when ID validity is uncertain to avoid incorrect selection counts.
      * \see selectByRect()
      * \see selectByExpression()
+     * \since QGIS 3.40
```

Comment:
The since is in the wrong place here:

```suggestion
     * \param validateIds (since QGIS 4.0) if TRUE, filters out invalid feature IDs before
     * selection. Set to FALSE (default) for best performance. Enable validation when calling from
     * Python or when ID validity is uncertain to avoid incorrect selection counts.
     * \see selectByRect()
     * \see selectByExpression()
```